app/helpers/download_helper.rb

```diff
@@ -0,0 +1,22 @@
+# Custom download links based on how MIT implements geoblacklight
+module DownloadHelper
+  # https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Presigner.html
+  def presign_this(document)
+    # for minio
+    if Rails.env.test? || Rails.env.development?
+      Aws.config[:s3] = { endpoint: ENV['MINIO_URL'], force_path_style: true }
+    end
+
+    signer = Aws::S3::Presigner.new
```

Comment:
We may want to move this to a initializer. This is proof of concept right not for discussion of the general solution.